File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcPropertyDefinition/IfcPropertySetDefinition/IfcPreDefinedPropertySet/IfcWindowPanelProperties/IfcWindowPanelProperties.py
class IfcWindowPanelProperties(IfcPreDefinedPropertySet):
    def __init__(self):
        self.OperationType: IfcWindowPanelOperationEnum = None
        self.PanelPosition: IfcWindowPanelPositionEnum = None
        self.FrameDepth: IfcPositiveLengthMeasure = None
        self.FrameThickness: IfcPositiveLengthMeasure = None
        self.ShapeAspectStyle: IfcShapeAspect = None

    # === WHERE CLAUSES ===
    # ApplicableToType : (EXISTS(SELF\IfcPropertySetDefinition.DefinesType[1])) AND ('IFC4X3_DEV_73740fe4.IFCWINDOWTYPE' IN TYPEOF(SELF\IfcPropertySetDefinition.DefinesType[1]))

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcPropertyDefinition/IfcPropertySetDefinition/IfcPreDefinedPropertySet/IfcWindowPanelProperties/IfcWindowPanelProperties.js
class IfcWindowPanelProperties extends IfcPreDefinedPropertySet {
    constructor() {
        /** @type {IFCWINDOWPANELOPERATIONENUM} */
        this.OperationType = null;
        /** @type {IFCWINDOWPANELPOSITIONENUM} */
        this.PanelPosition = null;
        /** @type {IFCPOSITIVELENGTHMEASURE} */
        this.FrameDepth = null;
        /** @type {IFCPOSITIVELENGTHMEASURE} */
        this.FrameThickness = null;
        /** @type {IFCSHAPEASPECT} */
        this.ShapeAspectStyle = null;
    }

    // === WHERE CLAUSES ===
    // ApplicableToType : (EXISTS(SELF\IfcPropertySetDefinition.DefinesType[1])) AND ('IFC4X3_DEV_73740fe4.IFCWINDOWTYPE' IN TYPEOF(SELF\IfcPropertySetDefinition.DefinesType[1]))
}
